hooks/tk-multi-publish2/basic/publish_movie.py

```diff
@@ -369,71 +519,71 @@ def _get_version_entity(self, item):
             return version_number
-        
+
         try:
             version_number = int(metadata)
         except ValueError:
             pass
-            
+
         return version_number
 
     def _unreal_asset_set_version(self, asset_path, version_number):
         asset = unreal.EditorAssetLibrary.load_asset(asset_path)
-        
+
         if not asset:
             return
-            
+
         engine = sgtk.platform.current_engine()
         tag = engine.get_metadata_tag("version_number")
-        
+
         unreal.EditorAssetLibrary.set_metadata_tag(asset, tag, str(version_number))
         unreal.EditorAssetLibrary.save_loaded_asset(asset)
-        
+
         # The save will pop up a progress bar that will bring the editor to the front thus hiding the publish app dialog
         # Workaround: Force all Shotgun dialogs to be brought to front
         engine = sgtk.platform.current_engine()
         for dialog in engine.created_qt_dialogs:
             dialog.raise_()
 
-        
-    def _unreal_render_sequence_to_movie(self, destination_path, unreal_map_path, sequence_path, movie_name):
+    def _unreal_render_sequence_with_sequencer(self, output_path, unreal_map_path, sequence_path):
         """
-        Renders a given sequence in a given level to a movie file
-        
-        :param destination_path: Destionation folder where to generate the movie file
-        :param unreal_map_path: Path of the Unreal map in which to run the sequence
-        :param sequence_path: Content Browser path of sequence to render
-        :param movie_name: Filename of the movie that will be generated
+        Renders a given sequence in a given level to a movie file with the Level Sequencer.
+
+        :param str output_path: Full path to the movie to render.
+        :param str unreal_map_path: Path of the Unreal map in which to run the sequence.
+        :param str sequence_path: Content Browser path of sequence to render.
         :returns: True if a movie file was generated, False otherwise
                   string representing the path of the generated movie file
         """
+        output_folder, output_file = os.path.split(output_path)
+        movie_name = os.path.splitext(output_file)[0]
+
         # First, check if there's a file that will interfere with the output of the Sequencer
         # Sequencer can only render to avi file format
-        output_filename = "{}.avi".format(movie_name)
-        output_filepath = os.path.join(destination_path, output_filename)
-
-        if os.path.isfile(output_filepath):
+        if os.path.isfile(output_path):
             # Must delete it first, otherwise the Sequencer will add a number in the filename
             try:
-                os.remove(output_filepath)
-            except OSError as e:
-                self.logger.debug("Couldn't delete {}. The Sequencer won't be able to output the movie to that file.".format(output_filepath))
+                os.remove(output_path)
```

Comment:
Suggest check for existence of this file in `validate()` and offer an action to [re]move the file to continue.

Reply:
Not sure. This is a limitation (or a feature) of the Level Sequencer which checks if the file exist and add a number to the file name if it does exist. The Movie Render queue does not that and just overwrite the existing file if it does exist. So this makes the two similar in terms of behavior. Since I didn't write the Level Sequencer code (I just moved things around), I don't know if this was done like this on purpose or not.
So I would be tempted to leave the code as is for the time being, and check if potential overwrites need to be flagged to the user, or not.

Reply:
👍